### ISSUE TITLE
Would you be interested in working UTs which could be run on host system?

### ISSUE THREAD

**Issue by biocomp** [ddemidov/ev3dev-lang-cpp#60]
I did a bit of refactoring here, and inject mock system via virtual calls instead of macros and file generation: https://github.com/biocomp/ev3dev-lang-cpp2/

I messed with cmake files and headers quite a lot, I guess I can simplify the change if there's an interest.

Also, there's a bug in the cache which I've found with the tests.

**Comment by ddemidov** [comment]
Yes, please open a PR with the changes. The fix for the caching bug should probably go to a separate PR.
